bin/yb-ctl

```diff
@@ -640,6 +652,28 @@ class ClusterOptions:
         if self.enable_ysql is None:
             self.enable_ysql = True
 
+        # -----------------------------------------------------------------------------------------
+        # Client protocol ports
+        # -----------------------------------------------------------------------------------------
+
+        self.ysql_port = None
```

Comment:
Hmm do we really need to save these as self.xxx_port? can't we just override the self.base_ports directly?

Reply:
self.base_ports is a two-level dictionary that is much less convenient to work with.